### ISSUE TITLE
Patterns modal pops up again after closing when rotating screen on mobile

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#67549]
### Description

When opening an empty page in the site editor on a phone, closing the "choose a pattern" modal, and then rotating the phone the modal reopens. 

I think this is because on a big enough phone (most phones are pretty big nowadays), the landscape mode is wide enough to get the desktop view, so it's switching between edit and mobile in the [pages route](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/site-editor-routes/pages.js), and the editor re-renders every time it does that, so the "closed" state of the modal gets reset.

### Step-by-step reproduction instructions

1. Open the Pages section of the site editor on a phone, and navigate to an empty page;
2. "Choose a pattern" modal should pop up. Close it;
3. Rotate the phone and see modal popping up again.

### Screenshots, screen recording, code snippet

https://github.com/user-attachments/assets/2262d1f0-c820-46bf-b524-d43d8d8984e8

### Environment info

latest Gutenberg trunk

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by Anuj-Rathore24** [comment]
Hey @tellthemachines, This issue is not replicable as you can see in my screen recording. Can you mention the versions you are using.

Version: 

WordPress: 6.7.1
Gutenberg: 19.8.0-rc.1

https://github.com/user-attachments/assets/9473999f-d849-4a6b-b7ed-58bc0fa9ff2d

**Comment by tellthemachines** [comment]
@Anuj-Rathore24 you're testing on too small a device (iPhone SE) so it doesn't hit the breakpoint at which the view is switched to desktop mode. Try testing on iPhone 12 Pro or Pixel 7.

**Comment by Anuj-Rathore24** [comment]
@tellthemachines, You are correct, I can see the following issue in iPhone 12 PRO!!

**Comment by Anuj-Rathore24** [comment]
Additionally, 

I am getting the following warning in the console,
```
index.js:177 wp-block-library-editor-css was added to the iframe incorrectly. Please use block.json or enqueue_block_assets to add styles to the iframe. 

<link rel="stylesheet" id="wp-block-library-editor-css" href="http://github-contribution.local/wp-content/plugins/gutenberg/build/block-library/editor.css?ver=1733378006" media="all">
```

**Comment by Mamaduka** [comment]
@Anuj-Rathore24, what version of Gutenberg are you using? A similar warning has been fixed recently - #66628.

**Comment by Anuj-Rathore24** [comment]
Yep its resolved, my automatic update was scheduled after 7 hours😅

**Comment by Mamaduka** [comment]
@Anuj-Rathore24, when testing bug reports, ensure you're running the latest Gutenberg versions.
